File: frontend/client-app/src/types/order/UserOrderResponse.ts
import type { PaymentResponse } from '@/types/payment/PaymentResponse'
import type { OrderItemResponse } from '@/types/order/OrderItemResponse'
import type { UserSelfResponse } from '../user/UserSelfResponse'

export interface UserOrderResponse {
  id: number
  orderDate: string
  status: string
  customer: UserSelfResponse
  payment?: PaymentResponse
  items: OrderItemResponse[]
}

File: backend/Marketplace/Marketplace.Core/DTOs/Order/UserOrderResponse.cs
using Marketplace.Core.DTOs.Payment;

namespace Marketplace.Core.DTOs.Order
{
    public class UserOrderResponse
    {
        public long Id { get; set; }
        public DateTime OrderDate { get; set; }
        public string Status { get; set; } = null!;
        public PaymentResponse? Payment { get; set; }
        public List<OrderItemResponse> Items { get; set; } = new();
    }
}
